Helper/Order.php

```diff
@@ -1644,15 +1644,20 @@ protected function isCaptureHookRequest($newCapture)
 
     /**
      * @param OrderInterface $order
-     * @param                                        $captureAmount
+     * @param float          $captureAmount
      *
+     * @return void
      * @throws \Exception
      */
-    protected function validateCaptureAmount(OrderInterface $order, $captureAmount) {
+    protected function validateCaptureAmount(OrderInterface $order, $captureAmount)
+    {
         $isInvalidAmount = !isset($captureAmount) || !is_numeric($captureAmount) || $captureAmount < 0;
-        $isInvalidAmountRange = $order->getTotalInvoiced() + $captureAmount > $order->getGrandTotal();
 
-        if($isInvalidAmount || $isInvalidAmountRange) {
+        $isInvalidAmountRange = $this->cartHelper->getRoundAmount($order->getTotalInvoiced())
```

Comment:
a case where this can fail even with this fix.

$captureAmount: 203.29
$order->getTotalInvoiced(): 
$order->getGrandTotal(): 203.2844
$captureAmount + $order->getTotalInvoiced(): 20329 > 20328 -> leads to raise an exception

Reply:
also fwiw i think it might be better tp add and then round to lose as little precision as possible?

Reply:
I believe in the case you've provided where `$order->getGrandTotal(): 203.2844` the captured amount will be 203.28, not 203.29. It's because of the operation of rounding will be done before the order information (including amounts) would be sent to Bolt. Then, when Bolt calls the order management webhook, it will be `$captureAmount: 203.28`

Reply:
what if

captureAmount = 203.49
getTotalInvoiced = 2.49
getGrandTotal = 205.9877

= (20349+249) > 20599?

i think it might just be safer to acount for edge cases than to say we send something different etc, because there are many ways to come with a case to break the above logic since you round before adding / checking?

Reply:
you might be right in terms of the operations taking place, but it is not obvious to a new person entering the codebase?

Reply:
Sorry, I've mistaken in the previous comment:
> the condition will remain `true`.

should be
> the condition will remain `false`, and no exception raises.

What about example with total invoiced, it would be:
> = (20349+249 = 20598) > 20599

In this case, the condition is also `false`, and it's not an error here :)

Finally, you're absolutely right about the lack of comments on what these rounding operations are applied for, and I will add it to the code.